core/src/hir/attrs.rs

```diff
@@ -772,6 +772,9 @@ pub struct BackendAttrSupport {
     pub iterables: bool,
     /// Marking a method as the `[]` operator, which is special in this language.
     pub indexing: bool,
+
+    /// Support for Option<Struct> and Option<Primitive>
+    pub option: bool,
```

Comment:
I recognize that this struct is called attrsupport and this is not an attribute, but it was always intended to be used for heavy new features as well. We could rename the struct.

In case of Option I _hope_ to make it supported in all backends (and required of future backends). In case of callbacks I'll probably keep the flag around long term.